### PR TITLE
errors: check for an error before prefixing

### PR DESCRIPTION
**Pull request by jlebon**
Minor tweak to the new `GLNX_AUTO_PREFIX_ERROR`. Since the common case
is that there's no errors, let's bring down the same check that
`g_prefix_error` does to avoid a function call most of the time.

### CODE REVIEW

File: glnx-errors.h [hunk 1]
```diff
@@ -109,7 +109,8 @@ typedef struct {
 static inline void
 glnx_cleanup_auto_prefix_error (GLnxAutoErrorPrefix *prefix)
 {
-  g_prefix_error (prefix->error, "%s: ", prefix->prefix);
+  if (prefix->error && *(prefix->error))
```
Comment (cgwalters) [inline glnx-errors.h:112]: Mmm...OK, I do random performance things too, but my basic rule is more like "avoid doing silly things inside loops".  We make *tons* of function calls into glib (and other libraries).  I haven't run a profiler on either ostree or rpm-ostree in a while (i'm suspecting you haven't either) and I'm certain doing so would find some absolutely silly stuff in places we didn't think of.

I'm fine with this patch if you prefer it, but I'd hesitate to do a lot more of this.  WDYT?

Comment (cgwalters) [reply glnx-errors.h:112]: It's definitely true that other cleanups do things like this; but it's also kind of silly, `g_autoptr_cleanup_generic_gfree()` also does a `NULL` check, then so does `g_free()`, then so does `free()`...

Comment (jlebon) [reply glnx-errors.h:112]: I think we should have a tag for these kinds of optimizations; those that don't actually make any practical difference in the big picture, but give the author/reviewer these warm fuzzy feelings knowing that something runs ever so slightly faster thanks to their insights, that something that used to take N CPU cycles, now takes N-5 cycles, where N > 100000000. A meditative relaxation exercise of sorts, allowing software engineers to commune with the craft they have come to learn and love. I'll call them "cozy patches".

This is a cozy patch. I'd say [this](https://github.com/projectatomic/rpm-ostree/pull/852#discussion_r124105791) is an even better cozy patch (the greater the micro-gains, the cozier it is). Monetarily 💰, it's not the most profitable way to spend one's time (or one's employer's time), but it's a great pick-me-up at the end of the day or when you need a breath of fresh air ☁️ between sessions of intense code-staring. 🖥 

I think my golden rule for cozy patches is that they should always be considered optional and never actively steal time 🕙 away from other, more serious, work. If they take more than 5 minutes to write and review (combined), then you've gone over. Any time more than that, it has to actually have a tangible performance benefit.

From now on, if I say "here is a cozy patch", then you'll know what I mean and apply the level of attention and urgency that it (doesn't) deserve(s). :)

**Comment by gnomesysadmins** [comment]
Thank you for contributing to libglnx!

libglnx uses Bugzilla for code review.

If you have never contributed to GNOME before make sure you have read the
getting started documentation:
http://www.gnome.org/get-involved

Otherwise please visit
https://wiki.gnome.org/Newcomers
and follow the instructions there to upload your change to Bugzilla.

File: glnx-errors.h (hunk 1 above)
Comment (cgwalters) [reply glnx-errors.h:112]: Hah, I like it.  Fair enough 😄

Comment (cgwalters) [reply glnx-errors.h:112]: Pushed, thanks!

**Pull request closed without merging**